src/main/java/com/google/jenkins/plugins/computeengine/CleanLostNodesWork.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.jenkins.plugins.computeengine;
+
+import com.google.api.services.compute.model.Instance;
+import hudson.Extension;
+import hudson.model.PeriodicWork;
+import hudson.model.Slave;
+import jenkins.model.Jenkins;
+import org.jenkinsci.Symbol;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import static com.google.common.collect.ImmutableMap.of;
+import static com.google.jenkins.plugins.computeengine.ComputeEngineCloud.CLOUD_ID_LABEL_KEY;
+import static java.util.Collections.emptyList;
+
+/**
+ * Periodically checks if there are no lost nodes in GCP.
+ * If it finds any they are deleted.
+ */
+@Extension
+@Symbol("cleanLostNodesWork")
+public class CleanLostNodesWork extends PeriodicWork {
+    protected final Logger logger = Logger.getLogger(getClass().getName());
+
+    /**
+     * {@inheritDoc}
+     */
+    @SuppressWarnings("unchecked")
+    @Override
+    public long getRecurrencePeriod() {
+        return HOUR;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @SuppressWarnings("unchecked")
+    @Override
+    protected void doRun() {
+        logger.log(Level.FINEST, "Starting clean lost nodes worker");
+        getClouds().forEach(this::cleanCloud);
+    }
+
+    private void cleanCloud(ComputeEngineCloud cloud) {
+        try {
+            logger.log(Level.FINEST, "Cleaning cloud " + cloud.getCloudName());
+            List<Instance> remoteInstances = findRemoteInstances(cloud);
+            Set<String> localInstances = findLocalInstances(cloud);
+            remoteInstances
+                    .stream()
+                    .filter(remote -> isOrphaned(remote, localInstances))
+                    .forEach(remote -> terminateInstance(remote, cloud));
+        } catch (Exception ex) {
```

Comment:
Suggest replacing "Exception" type with the types which actually get thrown.

Reply:
There is no known exceptions..

Reply:
So I either remove it or leave it as its..

Reply:
If that's the case, then let's remove the try/catch block.

Reply:
Removed..